helpers/withAuth.tsx

```diff
@@ -24,4 +27,83 @@ const withAuth = <P extends { children: React.ReactNode }>(WrappedComponent: Rea
   return Wrapper;
 };
 
+export const withAdminAuth = <P extends { children: React.ReactNode }>(WrappedComponent: React.ComponentType<P>) => {
+  const Wrapper: React.FC<P> = (props) => {
+    const [isPageLoading, setIsPageLoading] = useState(true);
+    const router = useRouter();
+    const { auth } = useAuth();
+    const { token } = useAuthRevalidate();
+
+    useEffect(() => {
+      // there is no token found in the localstorage
+      if (!isAuthenticated(token)) {
+        router.push('/access-denied');
```

Comment:
if there's no token in the localStorage then they should be redirected to  /login instead.